napari/components/add_layers_mixin.py

```diff
@@ -896,3 +900,44 @@ def _add_layer_from_data(
                 raise exc
 
         return layer
+
+    def _save_individual_layers_with_plugin(
+        self, plugin_name: str, path: str, layers: List[layers.Layer],
+    ):
+        """Save a list of layers to a path using the specified plugin.
+
+        Parameters
+        ----------
+        plugin_name : str
+            Name of the plugin to use for saving.
+        path : str
+            A filepath, directory, or URL (or a list of any) to open.
+        layers : List[layers.Layer]
+            List of layers to be saved.
+        """
+        for layer in layers:
+            write_layer_with_plugin(
+                plugin_name,
+                path,
+                data=layer.data,
+                meta=layer._get_state(),
+                layer_type=layer.__class__.__name__.lower(),
+            )
+
+    def _save_multiple_layers_with_plugins(
```

Comment:
@tlambert03 is it now strange that with this function you don't specify the plugin name but search through all possible ones?

Are we weirdly coupling the ability to search through and the ability to select an individual plugin and what we really want for all our hookspecs is an API where if we pass a plugin name it uses it and if we don't it searches through to try and find one that works? We could extend this to the readers too - i could imaging a dialog which let you pick which reader you wanted to use too

Reply:
> what we really want for all our hookspecs is an API where if we pass a plugin name it uses it and if we don't it searches through to try and find one that works

this is what I think we want, and yes I see it as analogous to reader plugins:  when they don't select a plugin, it loops through implementations and picks the first one that hits, as determined by the sort order.  When they _do_ select one, it tries that specific one and fails if it doesn't work (we still need to add that for readers, but it's always been [something we planned to do](https://github.com/napari/napari/pull/1023#issuecomment-595551417)).

i think part of the weirdness here is actually due to the previous `_save_individual_layers_with_plugin` function, which I don't think belongs.